### PR TITLE
Added collapsible GA and Preview in each artifact.

### PR DESCRIPTION
**Pull request by sima-zhu**
Before click:
image

After click:
image

Will do some clean up on the js file.

### CODE REVIEW

File: eng/docgeneration/Generate-DocIndex.ps1 [hunk 1]
```diff
@@ -98,9 +98,9 @@ function GenerateDocfxTocContent([Hashtable]$tocContent, [String]$lang) {
     $DocOutDir = "${RepoRoot}/docfx_project"
 
     LogDebug "Initializing Default DocFx Site..."
-    & $($DocFx) init -q -o "${DocOutDir}"
+    #& $($DocFx) init -q -o "${DocOutDir}"
     # The line below is used for testing in local
-    #docfx init -q -o "${DocOutDir}"
+    docfx init -q -o "${DocOutDir}"
```
Comment (sima-zhu) [inline eng/docgeneration/Generate-DocIndex.ps1:103]: For testing purpose, will revert the change once ready to merge

**Comment by weshaggard** [comment]
I was thinking something like:

```
azure-core
- <link to latest ga>
- <link to latest preview>
- collapsible group for other versions. 
```

File: eng/docgeneration/templates/matthews/styles/main.js [hunk 2]
```diff
@@ -78,24 +79,62 @@ function httpGetAsync(targetUrl, callback) {
 }
 
 function populateIndexList(selector, packageName) {
-    url = "https://azuresdkdocs.blob.core.windows.net/$web/" + SELECTED_LANGUAGE + "/" + packageName + "/versioning/versions"
-
-    httpGetAsync(url, function (responseText) {
-
-        var publishedversions = document.createElement("ul")
+    var url = "https://azuresdkdocs.blob.core.windows.net/$web/" + SELECTED_LANGUAGE + "/" + packageName + "/versioning/versions"
+    var latestGAUrl = "https://azuresdkdocs.blob.core.windows.net/$web/" + SELECTED_LANGUAGE + "/" + packageName + "/versioning/latest-ga"
```
Comment (weshaggard) [inline eng/docgeneration/templates/matthews/styles/main.js:83]: If GA or preview is missing lets just not show that list item. Also lets not add any text just have the version link.

File: eng/docgeneration/templates/matthews/styles/main.js [hunk 3]
```diff
@@ -78,24 +79,62 @@ function httpGetAsync(targetUrl, callback) {
...
+            version = responseText.match(/[^\r\n]+/g)
+            $(latestVersions).append('<li><a href="' + getPackageUrl(SELECTED_LANGUAGE, packageName, version) + '" target="_blank">' + 'Latest Preview Version: ' + version + '</a></li>')
         }
-
-        $(selector).after(publishedversions)
+        else {
+            $(latestVersions).append('<li>No latest Preview versions found.</li>')
+        }                
     })
+    var publishedVersions = $('<ul style="display: none;"></ul>')
```
Comment (weshaggard) [inline eng/docgeneration/templates/matthews/styles/main.js:104]: Lets see if we can make the other versions text align with the other list bullets.

Comment (sima-zhu) [reply eng/docgeneration/templates/matthews/styles/main.js:97]: Done.

File: eng/docgeneration/Generate-DocIndex.ps1 [hunk 4]
```diff
@@ -134,9 +134,9 @@ function GenerateDocfxTocContent([Hashtable]$tocContent, [String]$lang) {
     Copy-Item "$($RepoRoot)/CONTRIBUTING.md" -Destination "${DocOutDir}/api/CONTRIBUTING.md" -Force
 
     LogDebug "Building site..."
-    & $($DocFx) build "${DocOutDir}/docfx.json"
+    #& $($DocFx) build "${DocOutDir}/docfx.json"
```
Comment (weshaggard) [inline eng/docgeneration/Generate-DocIndex.ps1:137]: orthogonal but I'm curious why you don't just remove these and just pass docfx as a parameter?

**Review by weshaggard — APPROVED** [review]
Updated screen shots look nice. Changes look good lets get this deployed so we can see it in action.

File: eng/docgeneration/Generate-DocIndex.ps1 (hunk 4 above)
Comment (sima-zhu) [reply eng/docgeneration/Generate-DocIndex.ps1:137]: It is better to pass the docfx path in my local run. I am using nuget and npm command to install docfx commands. Have some trouble to find the right path. The best approach is to pass the path as an argument.
